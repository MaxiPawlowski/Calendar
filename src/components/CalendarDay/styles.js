import { createStyles } from '@material-ui/core';

export default () =>
  createStyles({
    root: {
      flex: `0 0 calc(${100 / 7}% - 10px)`,
      height: '100px',
      margin: '5px',
      padding: '5px',
      backgroundColor: '#e4f0f5',
      border: '1px solid rgba(50,50,50,.2)',
      borderRadius: '4px',
      display: 'flex',
      flexDirection: 'column',
      justifyContent: 'space-between'
    },
    button:{
      padding: '0px',
      justifyContent: 'flex-start',
      textTransform: 'inherit'
    },
    dotContainer:{
      display: 'flex',
      flexDirection: 'column',
      flexWrap: 'wrap',
      height: '30px',
      alignContent: 'baseline',
    },
    reminderDot: {
      height: '10px',
      width: '10px',
      borderRadius: '100%',
      marginRight: '5px',
      marginBottom: '5px',
    },
  });
