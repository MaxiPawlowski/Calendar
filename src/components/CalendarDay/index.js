import React from "react";
import { withStyles } from '@material-ui/styles';
import { Button, Typography} from '@material-ui/core';
import PropTypes from 'prop-types';
import DeleteForeverIcon from '@material-ui/icons/DeleteForever';
import styles from "./styles";

const CalendarDay = ({ classes, date, reminders, handleDayExpand, deleteReminder }) => (
  <div className={classes.root}>
    <Typography className={classes.typography}>
      {date.getDate()}
      {reminders.length > 0 && <DeleteForeverIcon className={classes.icon} onClick={() => deleteReminder(reminders)} />}
    </Typography>
    <div className={classes.dotContainer}>
      {reminders.map(({ color }, index) => (
        <div
          key={index}
          className={classes.reminderDot}
          style={{backgroundColor:color}}
        />
        ))}
    </div>
    {(reminders.length > 0) && (
    <Button className={classes.button} onClick={() => handleDayExpand({reminders, date})}>
          See More...
    </Button>
      )}
  </div>
);

CalendarDay.propTypes = {
  classes: PropTypes.shape({
    root: PropTypes.string.isRequired,
    typography: PropTypes.string.isRequired,
    button: PropTypes.string.isRequired,
    icon: PropTypes.string.isRequired,
    dotContainer: PropTypes.string.isRequired,
    reminderDot: PropTypes.string.isRequired,
  }).isRequired,

  date: PropTypes.instanceOf(Date).isRequired,

  reminders: PropTypes.arrayOf(PropTypes.shape({
    title: PropTypes.string.isRequired,
    color: PropTypes.string.isRequired,
    date: PropTypes.instanceOf(Date).isRequired,
    id: PropTypes.number.isRequired,
    city: PropTypes.oneOfType([
      PropTypes.string.isRequired,
      PropTypes.shape({
        description: PropTypes.string.isRequired,
        location: PropTypes.shape({
          lat: PropTypes.number.isRequired,
          lng: PropTypes.number.isRequired,
        }).isRequired
      }),
    ]),
  })).isRequired,

  handleDayExpand: PropTypes.func,
  deleteReminder: PropTypes.func,
};

CalendarDay.defaultProps = {
  handleDayExpand: () => {},
  deleteReminder: () => {},
}

export default withStyles(styles)(CalendarDay);
