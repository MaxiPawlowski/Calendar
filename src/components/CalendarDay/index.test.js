import React from 'react';
import { shallow } from 'enzyme';
import CalendarDay from "./index";

describe('CalendarDay', () => {
  let wrapper;

  describe('when rendered', () => {
    beforeAll(() => {
      wrapper = shallow(
        <CalendarDay
          reminders={[]}
          date={new Date(Date.now())}
        />
      );
    });

    it('should render correctly', () => {
      expect(wrapper.dive()).toMatchSnapshot();
    });
  });
});
