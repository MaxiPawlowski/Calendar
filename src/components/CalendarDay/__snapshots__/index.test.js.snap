// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarDay when rendered should render correctly 1`] = `
<div
  className="CalendarDay-root-1"
>
  14
  <div
    className="CalendarDay-dotContainer-3"
  />
</div>
`;
